client/components/map.jsx

```diff
@@ -40,13 +49,108 @@ class Mapbox extends React.Component {
         accessToken: mapboxgl.accessToken,
         mapboxgl: mapboxgl
       })
+        .on('result', result => {
+          this.setState({ inLocations: null, showFinish: false });
+          const locationName = result.result['place_name_en-US'];
+          if (!this.state.locations[0]) {
+            this.setState({
+              location: locationName,
+              inLocations: false
+            });
+          } else {
+            for (let i = 0; i < this.state.locations.length; i++) {
+              if (this.state.locations[i] === locationName) {
+                this.setState({
+                  location: locationName,
+                  inLocations: true
+                }, () => setTimeout(() => {
+                  this.setState({
+                    showFinish: true
+                  });
+                }, 2000));
+                break;
+              } else {
+                this.setState({
+                  location: locationName,
+                  inLocations: false
+                });
+              }
+            }
+          }
```

Comment:
Try to refactor this so that it's not calling `setState` over and over in the loop. I think you can just remove lines 72-77